tcms/report/views.py

```diff
@@ -81,73 +82,80 @@ def overview(request, product_id, template_name='report/overview.html'):
 
 class ProductVersionReport(TemplateView, ProductVersionReportData):
     template_name = 'report/version.html'
+    product = None
+    selected_version = None
 
-    def get(self, request, product_id):
+    def get(self, request, *args, **kwargs):
         try:
-            self.product = Product.objects.only('name').get(pk=product_id)
-
+            self.product = Product.objects.only('name').get(pk=kwargs.get('product_id'))
             version_id = request.GET.get('version_id')
             if version_id is not None:
                 qs = self.product.version.only('value')
                 self.selected_version = qs.get(pk=int(version_id))
         except (TypeError, ValueError, ObjectDoesNotExist) as error:
             raise Http404(error)
 
-        return super(ProductVersionReport, self).get(request, product_id)
+        return super().get(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
-        versions = self.product.version.only('product', 'value')
         product_id = self.product.pk
 
-        plans_subtotal = self.plans_subtotal(product_id)
-        running_runs_subtotal = self.runs_subtotal(product_id, True)
-        finished_runs_subtotal = self.runs_subtotal(product_id, False)
-        cases_subtotal = self.cases_subtotal(product_id)
-        case_runs_subtotal = self.case_runs_subtotal(product_id)
-        finished_case_runs_subtotal = \
-            self.finished_case_runs_subtotal(product_id)
-        failed_case_runs_subtotal = self.failed_case_runs_subtotal(product_id)
-
-        for version in versions:
-            vid = version.pk
-            version.plans_count = plans_subtotal.get(vid, 0)
-            version.running_runs_count = running_runs_subtotal.get(vid, 0)
-            version.finished_runs_count = finished_runs_subtotal.get(vid, 0)
-            version.cases_count = cases_subtotal.get(vid, 0)
-            version.failed_case_runs_count = failed_case_runs_subtotal.get(vid,
-                                                                           0)
-
-            m = finished_case_runs_subtotal.get(vid, 0)
-            n = case_runs_subtotal.get(vid, 0)
-            if m and n:
-                version.case_run_percent = round(m * 100.0 / n, 1)
-            else:
-                version.case_run_percent = .0
-
         case_runs_status_subtotal = None
         selected_version = getattr(self, 'selected_version', None)
         if selected_version is not None:
             case_runs_status_subtotal = self.case_runs_status_subtotal(
                 product_id, selected_version.pk)
 
-        data = super(ProductVersionReport, self).get_context_data(**kwargs)
+        data = super(ProductVersionReport, self).get_context_date(**kwargs)
```

Comment:
Fix typo in method name and always test. Perform manual tests for code which isn't covered by automated tests.

Reply:
Got it!